packages/community/anthropic/src/provider.ts

```diff
@@ -0,0 +1,416 @@
+
+export type AnthropicConnectionConfig = {
+  apiKey: string;
+};
+
+export class AnthropicLLMProvider extends BaseCompletionProvider<Anthropic> implements ICompletionProvider {
+  constructor(logger: Logger, provider: AIProviderConfig) {
+    super(logger, provider);
+  }
+
+  @Span('Anthropic.getMaxReplyTokens')
+  getMaxReplyTokens(request: CompletionRequest): number {
+    const maxTok = request.config?.max_tokens || 100; // REPLACE THIS!!
+    this.logger.log(maxTok, ' = maxTok');
+    return maxTok;
+  }
+
+  @Span('Anthropic.completion')
+  public async completion(
+    request: CompletionRequest,
+    connection: AIConnection<AnthropicConnectionConfig>,
+    model: ResourceModelConfig,
+    metadata: CompletionMetadata,
+    observable: Subject<CompletionResponse>,
+  ): Promise<CompletionResponse> {
+    //const callCompletion = this.errorHandling(this.callCompletion.bind(this));
+    return await this.callCompletion(request, connection, model, metadata, observable);
+  }
+
+  private async getRequestTokensDetails(
+    request: CompletionRequest,
+  ): Promise<{ usedTokens: number; completionUsedTokens: number; promptUsedTokens: number }> {
+    const usageClient = new TokenCounter({
+      model: 'cl100k_base',
+      messages: request.messages.map(({ toolCalls, ...msg }) => ({
+        ...msg,
+        role: msg.role,
+        content: msg.content ?? '',
+        tool_calls: toolCalls as TokenMessage['tool_calls'],
+      })),
+    });
+    const { usedTokens, completionUsedTokens, promptUsedTokens } = usageClient;
+    return { usedTokens, completionUsedTokens, promptUsedTokens };
+  }
+
+  @Span('Anthropic.getRequestTokens')
+  public async getRequestTokens(request: CompletionRequest): Promise<number> {
+    const { usedTokens } = await this.getRequestTokensDetails(request);
+    return usedTokens;
+  }
+
+  private async callCompletion(
+    request: CompletionRequest,
+    connection: AIConnection<AnthropicConnectionConfig>,
+    model: ResourceModelConfig,
+    metadata: CompletionMetadata,
+    observable: Subject<CompletionResponse>,
+  ): Promise<CompletionResponse> {
+    const client = await this.createClient(connection);
+
+    let toolChoice: ToolChoice = { type: 'auto' }; // default to auto
+    if (request.toolChoice?.auto) {
+      toolChoice = {
+        type: 'auto',
+      };
+    } else if (request.toolChoice?.none) {
+      toolChoice = {
+        type: 'any',
+      };
+    } else if (request.toolChoice?.tool && request.toolChoice.tool.function?.name) {
+      toolChoice = {
+        type: 'tool',
+        name: request.toolChoice.tool.function.name,
+      };
+    }
+
+    // Handle tool choice and streaming as per Anthropic API
+    const anthropicRequest: CallAPIRequest = {
+      ...(request.config || {}),
+      max_tokens: request?.config?.max_tokens || 1024, // <-- this should be a param in our ai provider setup, which can be overwritten by individual requests
```

Comment:
I think we should have a mapping between the max tokens for each Anthropic model, and not hard coded.

Reply:
I think it should be provided by the user in the console... So that they're aware that there's a max_tokens that gets set. Otherwise they don't know. This can be quite impactful to the AI answer..